dashboard/client/src/App.js

```diff
@@ -13,6 +13,16 @@ import 'bootstrap/dist/css/bootstrap.min.css';
 import './App.css';
 
 export class App extends Component {
+  routeOrRedirect() {
+    let cookie = document.cookie ? document.cookie : undefined
+
+    if (cookie) {
+      let decodedCookie = JSON.parse(base64Decode(document.cookie.split('.')[1]));
```

Comment:
Add range checking on the index, we can't guarantee there's > 0 elements, so in the case of a bad cookie that parsed, we'd cause an exception here.